client/src/utils/__mocks__/MockedSocketIO.js

```diff
@@ -1,21 +1,34 @@
 // @flow
-let EVENTS: {[string]: any} = {};
+
+type eventQueueT = {[string]: any}
+
+let EVENTS: eventQueueT = {};
+let EMITTED: eventQueueT = {};
+
+const push = (queue: eventQueueT, event: string, arg: any) => {
+    if (queue[event]) {
+      return queue[event].push(arg);
+    }
+    queue[event] = [arg];
+};
 
 function emit(event: string, ...args: any) {
   EVENTS[event].map(func => func(...args));
 }
 
 const socket = {
   on(event: string, func) {
-    if (EVENTS[event]) {
-      return EVENTS[event].push(func);
-    }
-    EVENTS[event] = [func];
+    push(EVENTS, event, func);
+  },
+  emit(event: string, data: any) {
+    push(EMITTED, event, data);
   },
-  emit,
   has(event: string) {
     return (EVENTS[event]) ? true : false;
   },
+  hasEmitted(event: string) {
```

Comment:
@beris147 take a look at this
I needed to test if an event was emitted, so I added this.
Do you think this is the best approach.
Another option is to have two symmetric objects, one for the client and another for the server, with individual event queues

Reply:
Looks good to me, this will only be used for testing, so if in the future we need to do more specific stuff we can change it.